packages/app/src/sections/dashboard/Stake/RewardsTab.tsx

```diff
@@ -0,0 +1,387 @@
+				title: t('dashboard.rewards.snx-rewards.title'),
+				copy: t('dashboard.rewards.snx-rewards.copy'),
+				labels: [
+					{
+						label: t('dashboard.stake.portfolio.rewards.title'),
+						value: formatNumber(snxOpRewards, { minDecimals: 4 }),
+						valueIcon: <OptimismLogo height={18} width={18} />,
+					},
+				],
+				info: [
+					{
+						label: t('dashboard.rewards.estimated'),
+						value: NO_VALUE,
+					},
+				],
+			},
+		],
+		[
+			estimatedKwentaRewards,
+			estimatedOpRewards,
+			futuresFeePaid,
+			kwentaRewards,
+			opRewards,
+			period,
+			ratio,
+			snxOpRewards,
+			t,
+			totalFuturesFeePaid,
+		]
+	)
+
+	const handleChangeEpoch = useCallback(
+		(value: EpochValue) => () => {
+			dispatch(setSelectedEpoch(value.period))
+		},
+		[dispatch]
+	)
+
+	const formatOptionLabel = useCallback(
+		(option: EpochValue) => (
+			<div onClick={handleChangeEpoch(option)}>
+				<SelectLabelContainer>{option.label}</SelectLabelContainer>
+			</div>
+		),
+		[handleChangeEpoch]
+	)
+
+	return (
+		<RewardsTabContainer>
+			<HeaderContainer>
+				<StyledHeading variant="h4">{t('dashboard.rewards.title')}</StyledHeading>
+				<StakingSelect
+					formatOptionLabel={formatOptionLabel}
+					controlHeight={33}
+					options={epochData.sort((a, b) => b.period - a.period)}
+					optionPadding="0px"
+					value={selectedEpoch}
+					menuWidth={110}
+					components={{ IndicatorSeparator, DropdownIndicator }}
+					isSearchable={false}
+					variant="flat"
+					isDisabled={!isL2}
+				/>
+			</HeaderContainer>
+			<CardsContainer>
+				{rewardsInfo.map(({ key, title, copy, labels, info }) => (
+					<CardGrid key={key}>
+						<div>
+							<Body size="large" color="primary">
+								{title}
+							</Body>
+							<Body color="secondary">{copy}</Body>
+						</div>
+						<RewardsContainer>
+							<FlexDivRow justifyContent="flex-start" columnGap="25px">
+								{labels.map(({ label, value, labelIcon, valueIcon }) => (
+									<FlexDivCol rowGap="5px">
+										<IconContainer color="secondary">
+											{label}
+											{labelIcon}
+										</IconContainer>
+										<IconContainer size="large" color="preview">
+											{value}
+											{valueIcon}
+										</IconContainer>
+									</FlexDivCol>
+								))}
+							</FlexDivRow>
+							<FlexDivRow justifyContent="flex-start" columnGap="25px">
+								{info.map(({ label, labelIcon, value, valueIcon }) => (
+									<FlexDivCol rowGap="5px">
+										<IconContainer color="secondary">
+											{label}
+											{labelIcon}
+										</IconContainer>
+										<IconContainer size="large" color="primary">
+											{value}
+											{valueIcon}
+										</IconContainer>
+									</FlexDivCol>
+								))}
+							</FlexDivRow>
+						</RewardsContainer>
+					</CardGrid>
+				))}
+				<ButtonContainer>
+					<Button
+						variant="yellow"
+						size="small"
+						textTransform="uppercase"
+						isRounded
+						disabled={claimDisabledAll}
```

Comment:
Would be best to disable also when the transaction is pending and ideally show a loading state.

Reply:
Yes. Fixed. That's good practice. I will check the other buttons' state later.